src/components/helpers/AuthenticatorHelper.ts

```diff
@@ -0,0 +1,41 @@
+import OneWelcomeSdk, {Types} from 'onewelcome-react-native-sdk';
+
+const getAllAuthenticators = async (
+  setRegisteredAuthenticators: (authenticators: Types.Authenticator[]) => void,
+  setAllAuthenticators: (authenticators: Types.Authenticator[]) => void,
+  setPreferred: (authenticator: Types.Authenticator) => void,
+) => {
+  const profile = await OneWelcomeSdk.getAuthenticatedUserProfile();
+  const registeredAuthenticators =
+    await OneWelcomeSdk.getRegisteredAuthenticators(profile.profileId);
+
+  const allAuthenticators = await OneWelcomeSdk.getAllAuthenticators(
+    profile.profileId,
+  );
+
+  registeredAuthenticators.forEach(it => {
+    if (it.isPreferred) {
+      setPreferred(it);
+    }
+  });
+  setRegisteredAuthenticators(registeredAuthenticators);
+  setAllAuthenticators(allAuthenticators);
+};
+
+const setPreferredAuthenticator = async (
+  preferred: Types.Authenticator,
+  setMessage: (msg: string) => void,
+) => {
+  try {
+    const profile = await OneWelcomeSdk.getAuthenticatedUserProfile();
+    await OneWelcomeSdk.setPreferredAuthenticator(
+      profile.profileId,
+      preferred.id,
+    );
+    setMessage('The ' + preferred.name + ' is set');
```

Comment:
so the messages would become for example; "The TouchId / fingerprint is set". I would maybe make it a bit more human readable in the form of: `The Authenticator "${preferred.name}" is now set as the default authentication method`

Also string templates might be nicer ;)